### PR TITLE
Added Favorites Functionality

### PR DESCRIPTION
**Pull request by redliz5808**

### CODE REVIEW

File: src/components/CollectionPhotos/CollectionPhotos.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  componentDidMount() {
+    const { collectionid } = this.props;
+    this.retrieveCollectionPhotos(collectionid);
+  }
+
+  render() {
+    const { data } = this.state;
+    const readyWithPhotos = data;
```
Comment (yoloOnTheBattlefield) [inline src/components/CollectionPhotos/CollectionPhotos.js]: no loading 😅😅😅😅😅😅😅😅😅😅😅😅

File: src/pages/Collection/Collection.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            <Tags>
+              {tagsAvailable &&
+                data.tags.map((tag, index) => {
+                  if (index < 6) {
+                    return (
+                      <TagLink key={tag.title} to={`/search/${tag.title}`}>
+                        {tag.title}
+                      </TagLink>
+                    );
+                  }
```
Comment (yoloOnTheBattlefield) [inline src/pages/Collection/Collection.js]:
```
data.tags
  .filter((tag, index) => index < 6)
  .map((tag) => (
        <TagLink key={tag.title} to={`/search/${tag.title}`}>
          {tag.title}
        </TagLink>
    )
  )}

File: src/pages/Favorites/Favorites.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      try {
+        this.setState({ isLoading: true });
+        const { data } = await axios(
+          `${process.env.REACT_APP_API_BASE_URL}/photos/${photo}?client_id=${process.env.REACT_APP_API_KEY}`
+        );
+        this.setState({
+          photos: [...this.state.photos, data],
+          isLoading: false,
+        });
+        console.log(this.state.photos);
```
Comment (yoloOnTheBattlefield) [inline src/pages/Favorites/Favorites.js]: 🙄

File: src/pages/Home/Home.js [hunk 4]
```diff
@@ -98,9 +117,27 @@ class Home extends React.Component {
                           />
                         </StyledLink>
                         <StyledLink to={`/user/${value.user.username}`}>
-                          <UserName username={value.user.name} />
+                          <div>{value.user.name}</div>
                         </StyledLink>
-                        <Icon icon={<FaHeart />} stats={value.likes} />
+                        {Object.values(this.state.favoritePhotos).find((id) => {
+                          return id === value.id;
+                        }) ? (
```
Comment (yoloOnTheBattlefield) [inline src/pages/Home/Home.js]: but if `this.state.favoritePhotos` is an object, you don't need to convert it to an array. You can just do `this.state.favoritePhotos[value.id]`

File: src/pages/Photo/Photo.js [hunk 5]
```diff
@@ -59,20 +76,35 @@ class Photo extends React.Component {
             </StyledLink>
             <MainImage src={data.urls.regular} alt={data.alt_description} />
             <StyledDiv>
-              <Icon icon={<FaHeart />} stats={data.likes} />
+              {Object.values(this.state.favoritePhotos).find((id) => {
+                return id === data.id;
+              }) ? (
```
Comment (yoloOnTheBattlefield) [inline src/pages/Photo/Photo.js]: same here

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/pages/Favorites/Favorites.js [hunk 6]
```diff
@@ -1,52 +1,52 @@
...
-            return (
-              <ImageContainer>
-                <StyledLink to={`/photo/${photo.id}`}>
-                  <img src={photo.urls.small} alt={photo.description} />
-                </StyledLink>
-              </ImageContainer>
-            );
-          })}
-      </Container>
+      <Paper square>
```
Comment (yoloOnTheBattlefield) [inline src/pages/Favorites/Favorites.js:23]: Paper? https://gph.is/g/ZY6PG14

**Review by yoloOnTheBattlefield — APPROVED** [review]

File: src/pages/Favorites/Favorites.js (hunk 6 above)
Comment (redliz5808) [reply src/pages/Favorites/Favorites.js:23]: HAHA! It's what comes with the package for making the tabs. 🤷‍♀️

**Pull request merged**